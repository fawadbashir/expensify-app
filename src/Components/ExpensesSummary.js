import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'

export  const ExpensesSummary = (props) => {
    const Total = numeral(expensesTotal(props.expenses) / 100).format('$0,0.00')
    const expenseCount = props.expenses.length
    const expenseWord = expenseCount === 1 ? 'Expense' : 'Expenses'
    
    return(
        <div><h4>Viewing {props.expenses.length} {expenseWord} of Total {Total}</h4>
            
        </div>
    )
} 


const mapStateToProps = (state) =>{
   return {
       expenses : selectExpenses(state.expenses,state.filters)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)