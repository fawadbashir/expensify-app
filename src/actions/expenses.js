import database from '../firebase/firebase'

//ADD_EXPENSE
export const addExpense = (expense) => ({
  type: "ADD_EXPENSE",
  expense
})

export const startAddExpense = (expenseData = {}) => {
  return async (dispatch) => {
    const {
      description = "",
      note = "",
      amount = 0,
      createdAt = 0
    } = expenseData
    const expense = { description, note, amount, createdAt }

    const ref = await database.ref('expenses').push(expense)
    dispatch(addExpense({
      id: ref.key,
      ...expense
    }))
  }
}

//REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
})

 const startRemoveExpense = ({id} = {}) => {
  return async (dispatch) => { 
 await database.ref(`expenses/${id}`).remove()
     dispatch(removeExpense({id}))
  }

}

//EDIT_EXPENSE
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates
})

//SET_EXPENSE
export const setExpenses = (expenses) => ({
  type: 'SET_EXPENSES',
  expenses
})

export const startSetExpenses = () => {
  return  async (dispatch) => {
   const snapshot = await database.ref('expenses').once('value')
    const expenses = []
    snapshot.forEach((childSnapshot) => {
      expenses.push({
        id: childSnapshot.key,
        ...childSnapshot.val()
      })
    })
    dispatch(setExpenses(expenses))
  }
}

export { startRemoveExpense }