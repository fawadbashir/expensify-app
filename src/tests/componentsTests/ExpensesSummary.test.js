import React from 'react'
import { shallow } from 'enzyme'
import {createSerializer} from 'enzyme-to-json';
import { ExpensesSummary } from '../../components/ExpensesSummary.js'
import expenses from '../fixtures/expensesFixture'

expect.addSnapshotSerializer(createSerializer({mode: 'shallow'}));

test('should correctly render ExpensesSummary with single Expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses = {[expenses[0]]}/>)
    expect(wrapper).toMatchSnapshot()
})


test('should correctly render ExpensesSummary with multiple Expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses = {[expenses]}/>)
    expect(wrapper).toMatchSnapshot()
})

