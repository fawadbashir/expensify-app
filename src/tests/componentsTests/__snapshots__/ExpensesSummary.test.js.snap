// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with multiple Expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    Expense
     of Total 
    $0.00
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with single Expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    Expense
     of Total 
    $1.00
  </h1>
</div>
`;
